File: ruby/TensionsBetweenDifferentOOPrinciples/TensionBetweenSRPandLowCoupling/After/plateau.rb
module TensionBetweenSRPandLowCoupling
  module After
    class Plateau
      # Plateau class represents a plateau with well defined boundaries.
      # So Plateau class follows the Single Responsibility Principle (SRP).

      def initialize
        @plateau = [11, 13]
      end

      def contains(position)
        ((0 <= position.first || position.first <= @plateau.first) ||
        (0 <= position.last || position.last <= @plateau.last))
      end
    end
  end
end

File: Java/tensionsBetweenDifferentOOPrinciples/src/main/java/tensionsBetweenDifferentOOPrinciples/tensionBetweenSRPandLowCoupling/after/Plateau.java
package tensionsBetweenDifferentOOPrinciples.tensionBetweenSRPandLowCoupling.after;

import tensionsBetweenDifferentOOPrinciples.tensionBetweenSRPandLowCoupling.Tuple;

public class Plateau {
    // Plateau class represents a plateau with well defined boundaries.
    // So Plateau class follows the Single Responsibility Principle (SRP).

    private final Tuple<Integer, Integer> _plateau = new Tuple<>(11, 13);

    public boolean Contains(Tuple<Integer, Integer> position) {
        return ((0 <= position.getItem1() || position.getItem1() <= _plateau.getItem1()) ||
                (0 <= position.getItem2() || position.getItem2() <= _plateau.getItem2()));
    }
}

